### DISCUSSION TITLE
Exercise 14: atomic commit (7.3)

### DISCUSSION THREAD

**Discussion by dag7dev** [sapienzastudentsnetwork/distributed-systems#39] [Exercises]
Consider 2PC with the cooperative termination protocol. Describe a scenario (a particular execution) involving site failures only, which causes operational sites to become blocked.

**Comment by susannacifani** [comment]
I don't think the solution on the Drive is correct because it doesn't consider the cooperative termination protocol.
In fact, the solution says that the operational sites cannot proceed without the coordinator's decision and as a result they remain locked, waiting for a resolution that never arrives due to the coordinator's failure, but with the cooperative termination protocol we have that the participants know each other and exchange messages directly (this protocol mitigates the problem described in the solution on Drive, that considers 2PC without this protocol)

**Comment by Tommaso-Sgroi** [comment]
> I don't think the solution on the Drive is correct because it doesn't consider the cooperative termination protocol. In fact, the solution says that the operational sites cannot proceed without the coordinator's decision and as a result they remain locked, waiting for a resolution that never arrives due to the coordinator's failure, but with the cooperative termination protocol we have that the participants know each other and exchange messages directly (this protocol mitigates the problem described in the solution on Drive, that considers 2PC without this protocol)

The solution of this exercise is written in section 7.4 of the atomic commit pdf.

**Comment by Tommaso-Sgroi** [comment]
![image](https://github.com/sapienzastudentsnetwork/distributed-system-2023-2024/assets/62346799/a087dd08-c63a-4bed-97bf-4f112eeb1e57)

### Solution
![image](https://github.com/sapienzastudentsnetwork/distributed-system-2023-2024/assets/62346799/f3801a50-c372-45de-abff-7e60feaa7b73)
